### PR TITLE
Print module - implement module display for remaining modules (part2)

### PR DESCRIPTION
**Pull request by antimora**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [ ] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

#1357

### Changes

Added the module displays for the following modules:

1. gelu
2. loss
   - mse
   - huber
   - binary_cross_entropy
   - cross_entropy
3. dropout
4. leaky_relu
5. pool
   - adaptive_avg_pool1d
   - adaptive_avg_pool2d
   - avg_pool1d
   - avg_pool2d
   - max_pool1d
   - max_pool2d
6. norm
   - layer
   - rms
   - instance
   - group
   - batch
7. prelu
8. pos_encoding
9. relu
10. tanh
11. unfold
12. embedding
13. attention
    - mha
14. swiglu
15. conv
    - conv_transpose1d
    - conv_transpose2d
    - conv1d
    - conv2d
16. linear
17. rnn
    - lstm
    - gru
18. transformer
    - pwff
    - encoder
    - decoder
19. rope_encoding

### Testing

Added rendering unit tests for each module.

### CODE REVIEW

File: crates/burn-core/src/nn/transformer/pwff.rs [hunk 1]
```diff
@@ -36,13 +36,32 @@ pub struct PositionWiseFeedForwardConfig {
 ///
 /// Should be created using [PositionWiseFeedForwardConfig]
 #[derive(Module, Debug)]
+#[module(custom_display)]
 pub struct PositionWiseFeedForward<B: Backend> {
     linear_inner: Linear<B>,
     linear_outer: Linear<B>,
     dropout: Dropout,
     gelu: Gelu,
```
Comment (nathanielsimard) [inline crates/burn-core/src/nn/transformer/pwff.rs:44]: We should put the parameters public.

**Review by nathanielsimard — APPROVED** [review]
Only one comment, approving in advance! 🙏

File: crates/burn-core/src/nn/transformer/pwff.rs (hunk 1 above)
Comment (antimora) [reply crates/burn-core/src/nn/transformer/pwff.rs]: Fixed. I made other modules' attributes public for consistency since it's been requested frequently.

**Review by laggui — APPROVED** [review]
LGTM! Thanks for also fixing the modules parameters and adding docstrings 🙏

**Pull request merged**